### ISSUE TITLE
java.lang.NoSuchMethodError in version 1.2

### ISSUE THREAD

**Issue by shpoont** [iwarapter/sonar-puppet#258]
Versions:
Sonar: 5.3
SonarQube Scanner: 2.5

```
--- skip ---
INFO: Sensor QProfileSensor (done) | time=7ms
INFO: Sensor PuppetSquidSensor
Exception in thread "main" java.lang.NoSuchMethodError: org.sonar.api.component.ResourcePerspectives.as(Ljava/lang/Class;Lorg/sonar/api/component/Component;)Lorg/sonar/api/component/Perspective;
    at com.iadams.sonarqube.puppet.checks.ProjectChecks.addIssue(ProjectChecks.java:146)
    at com.iadams.sonarqube.puppet.checks.ProjectChecks.checkTestsDirectoryPresent(ProjectChecks.java:130)
    at com.iadams.sonarqube.puppet.checks.ProjectChecks.checkTestsDirectoryPresent(ProjectChecks.java:135)
    at com.iadams.sonarqube.puppet.checks.ProjectChecks.checkTestsDirectoryPresent(ProjectChecks.java:135)
    at com.iadams.sonarqube.puppet.checks.ProjectChecks.checkTestsDirectoryPresent(ProjectChecks.java:135)
    at com.iadams.sonarqube.puppet.checks.ProjectChecks.reportProjectIssues(ProjectChecks.java:64)
    at com.iadams.sonarqube.puppet.PuppetSquidSensor.save(PuppetSquidSensor.java:138)
    at com.iadams.sonarqube.puppet.PuppetSquidSensor.analyse(PuppetSquidSensor.java:118)
    at org.sonar.batch.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:58)
    at org.sonar.batch.phases.SensorsExecutor.execute(SensorsExecutor.java:50)
    at org.sonar.batch.phases.PhaseExecutor.execute(PhaseExecutor.java:98)
    at org.sonar.batch.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:185)
    at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:132)
    at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:117)
    at org.sonar.batch.scan.ProjectScanContainer.scan(ProjectScanContainer.java:243)
    at org.sonar.batch.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:238)
    at org.sonar.batch.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:228)
    at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:132)
    at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:117)
    at org.sonar.batch.task.ScanTask.execute(ScanTask.java:55)
    at org.sonar.batch.task.TaskContainer.doAfterStart(TaskContainer.java:86)
    at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:132)
    at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:117)
    at org.sonar.batch.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:122)
    at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:119)
    at org.sonar.runner.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:67)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.sonar.runner.impl.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:61)
    at com.sun.proxy.$Proxy0.execute(Unknown Source)
    at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:274)
    at org.sonar.runner.api.EmbeddedRunner.runAnalysis(EmbeddedRunner.java:165)
    at org.sonar.runner.api.EmbeddedRunner.runAnalysis(EmbeddedRunner.java:152)
    at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:118)
    at org.sonarsource.scanner.cli.Main.execute(Main.java:80)
    at org.sonarsource.scanner.cli.Main.main(Main.java:66)
```

**Comment by iwarapter** [comment]
Sneaky feeling this has already been fixed and is in the yet-to-be-released 1.3, will confirm.

**Comment by iwarapter** [comment]
Confirmed, this will be fixed in the next release :)

**Comment by shpoont** [comment]
Thanks, when do you plan to release? :)

**Comment by iwarapter** [comment]
i'll publish a rc tonight, providing all is well full version will be published 3 days later :)

**Comment by shpoont** [comment]
Thanks a lot! :+1:

**Comment by iwarapter** [comment]
Release candidate is available: https://github.com/iwarapter/sonar-puppet/releases/tag/1.3-RC1

**Comment by iwarapter** [comment]
Also dont forget to vote: https://groups.google.com/forum/#!topic/sonarqube/qLPb2CwhiWc

**Comment by iwarapter** [comment]
Fixed with the release of 1.3
